src/main/java/com/ssafy/trip/member/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.ssafy.trip.member.service;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.ssafy.trip.common.exception.ErrorCode;
+import com.ssafy.trip.common.exception.custom.BadRequestException;
+import com.ssafy.trip.common.exception.custom.ConflictException;
+import com.ssafy.trip.common.exception.custom.NotFoundException;
+import com.ssafy.trip.member.domain.Member;
+import com.ssafy.trip.member.repository.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class MemberServiceImpl implements MemberService {
+    private final MemberRepository memberRepository;
+    private final AmazonS3 s3Client;
+
+    @Value("${aws-adk.bucketName}")
+    private String bucketName;
+
+    @Transactional(readOnly = true)
+    public Member findById(Long id) {
+        return memberRepository.findByIdAndDeletedAtIsNull(id).orElseThrow(() ->
+                new NotFoundException(ErrorCode.MEMBER_NOT_FOUND));
+    }
+
+    @Override
+    public void modifyMember(Long id, MultipartFile imageFile, String nickname) {
+        Member member = memberRepository.findByIdAndDeletedAtIsNull(id).orElseThrow(() ->
+                new NotFoundException(ErrorCode.MEMBER_NOT_FOUND));
+
+        if (nickname != null && !member.getNickname().equals(nickname)) {
+            if (memberRepository.existsByNickname(nickname)) {
+                throw new ConflictException(ErrorCode.NICKNAME_ALREADY_EXIST);
+            }
+            member.updateNickname(nickname);
+        }
+
+        nickname = member.getNickname();
+        if (imageFile != null) {
+            member.updateProfileUrl(uploadImage(nickname, imageFile));
+        }
```

Comment:
51번 줄을 if문 block 안에 넣거나, 변수로 값을 재할당하지 않고 바로 넣으면 좋을 것 같아요.

```suggestion
        if (imageFile != null) {
            nickname = member.getNickname();
            member.updateProfileUrl(uploadImage(nickname, imageFile));
        }
```
```suggestion
        if (imageFile != null) {
            member.updateProfileUrl(uploadImage(member.getNickname(), imageFile));
        }
```